src/interfaces/coral_web/src/hooks/chat.ts

```diff
@@ -248,9 +252,59 @@ export const useChat = (config?: { onSend?: (msg: string) => void }) => {
               break;
             }
 
-            case StreamEvent.TOOL_INPUT: {
-              const data = eventData.data as StreamToolInput;
-              toolEvents.push(data);
+            case StreamEvent.TOOL_CALLS_CHUNK: {
+              const data = eventData.data as StreamToolCallsChunk;
+
+              // Initiate an empty tool event if one doesn't already exist at the current index
+              const toolEvent: StreamToolCallsGeneration = toolEvents[currentToolEventIndex] ?? {
+                text: '',
+                tool_calls: [],
+              };
+              toolEvent.text += data?.text ?? '';
+
+              // A tool call needs to be added/updated if a tool call delta is present in the event
+              if (data?.tool_call_delta) {
+                const currentToolCallsIndex = data.tool_call_delta.index ?? 0;
+                let toolCall = toolEvent.tool_calls?.[currentToolCallsIndex];
+                if (!toolCall) {
+                  toolCall = {
+                    name: '',
+                    parameters: {},
+                  };
+                  toolCallParamaterStr = '';
+                }
+
+                if (data?.tool_call_delta?.name) {
+                  toolCall.name = data.tool_call_delta.name;
+                }
+                if (data?.tool_call_delta?.parameters) {
+                  toolCallParamaterStr += data?.tool_call_delta?.parameters;
+
+                  // Attempt to parse the partial parameter string as valid JSON to show that the parameters
+                  // are streaming in. To make the partial JSON string valid JSON after the object key comes in,
+                  // we naively try to add `"}` to the end.
+                  try {
+                    const partialParams = JSON.parse(toolCallParamaterStr + `"}`);
+                    toolCall.parameters = partialParams;
+                  } catch (e) {
+                    // Ignore parsing error
```

Comment:
should we report this somewhere to model/BE can fix common issues

Reply:
i don't think we have any reporting structure as of now 😅 but this should self-heal in the `TOOL_CALLS_GENERATION` event if anything. this is also mostly only in place because of the fact that we're receiving partial JSON strings so the appending of `"}` is guaranteed to work only while the param value is being streamed in and not any other part of the JSON. I have a feeling if there is a problem with the JSON itself it'll probably happen in `TOOL_CALLS_GENERATION` and it'll make the stream return early with an error.